src/index.js

```diff
@@ -165,89 +165,103 @@ exports.createFromPubKey = function (key, callback) {
     throw new Error('callback is required')
   }
 
-  let buf = key
-  if (typeof buf === 'string') {
-    buf = Buffer.from(key, 'base64')
-  }
+  try {
+    let buf = key
+    if (typeof buf === 'string') {
+      buf = Buffer.from(key, 'base64')
+    }
 
-  const pubKey = crypto.keys.unmarshalPublicKey(buf)
+    if (!Buffer.isBuffer(buf)) throw new Error('Supplied key is neither a base64 string nor a buffer')
 
-  pubKey.hash((err, digest) => {
-    if (err) {
-      return callback(err)
-    }
+    const pubKey = crypto.keys.unmarshalPublicKey(buf)
```

Comment:
what about wrapping this line in a try/catch block?

Reply:
The whole thing is in try catch. Also included the Buffer.from because it throw an error when a non-base64 string is passed to it.